teps/0091-trusted-resources.md

```diff
@@ -364,26 +373,70 @@ type KeylessRef struct {
 
 ### Configuration
 
-Reconciler will respond to verification failure based on ConfigMap settings to:
-1) Return error in reconciler, mark the status as error with `FailedVerification` reason;
-2) Not return error, but mark the TaskRun/PipelineRun as `FailedVerification`;
-3) Skip the verification;
-
-Configuration sample:
-```yaml
+ ```yaml
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: feature-flag
   namespace: tekton-pipelines
 data:
-  verification-policy: enforce
+  no-match-policy: "deny"
 ```
 
-`verification-policy`. (Optional, `enforce`, `warn` or `skip`, default `skip`):
- * `enforce`: Failing verification will mark the taskruns/pipelineruns as failed.
- * `warn`: Log warning but don't fail the taskruns/pipelineruns.
- * `skip`: Directly skip the verification.
+`no-match-policy`. (Optional, `allow`, `deny` or `warn`, default `allow`):
+ * `deny`: Reject the resource and fail the taskrun/pipelinerun if no matching policies are found.
+ * `allow`: Allow the resource if no matching policies are found.
+ * `warn`: Allow the resource and log a warning if no matching policies are found.
+
+ Reconciler will respond to verification failure based on ConfigMap and Verification Policy exsitence settings to:
+1) Return error in reconciler, mark the status as error with `FailedVerification` reason if `enforce` policies fail
+2) Not return error, but warn the log and update the status condition with `FailedVerification` reason if only found `warn` policies.
+3) Skip the verification if no policies exist and `no-match-policy` is set to `allow`.
+
+ **Note:**
+ * If the `no-match-policy` is set to `deny`, resources that do not match any policies will be rejected. Resources that match policies will be accepted if all the matching `enforce` policies pass or only match `warn` policies.
+
+ * If the `no-match-policy` is set to `allow` or `warn`, resources that do not match any policies will be accepted, and will emit warning log if set to `warn`. Resources that match policies will be accepted if all the matching `enforce` policies pass or only match `warn` policies.
+
+
+ #### Condition Update
+
+ Trusted resources should also update the taskrun/pipelinerun’s condition to indicate if it passes verification or not. This can be done via knative api.
+
+Cases:
+
+1. All matching policies pass verification
+Sample Condition:
+ ```go
+ apis.Condition{
+			Type:    apis.ConditionType(apis.ConditionSeverityInfo),
+			Status:  corev1.ConditionTrue,
+			Reason:  podconvert.ReasonResourceVerificationSuccess
+			Message: "Trusted resource verification passed",
+		}
+ ```
+
+ 2. `enforce` policy fails verification.
+Sample Condition:
+ ```go
+ apis.Condition{
+			Type:    apis.ConditionType(apis.ConditionSeverityInfo),
```

Comment:
I'm a bit confused by this Type, my understanding for our existing execution status is that we use apis.ConditionSucceeded, what is apis.ConditionSeverityInfo intended to be used for, and do we want to use that or do we want to use something custom? (would be useful to provide more detail around how we chose this type and what other options would be)

Reply:
Sure! If we use `apis.ConditionSucceeded` then it will be overwritten when taskrun is completed, using a different type can avoid this. The type in this example is from knative:
https://github.com/knative/pkg/blob/980a33719a10024f45e44320316c5bd35cef18d6/apis/condition_types.go#L52

The condition in taskrun status will be:
```
  - lastTransitionTime: "2023-03-01T18:17:05Z"
    message: Trusted resource verification passed
    reason: Verification Success
    status: "True"
    type: Info
```

I didn't think about creating our own condition type before but that could be another option. If we have a custom type then we could also avoid overwritten of the same type